xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-include/src/test/java/org/xwiki/rendering/internal/macro/IncludeMacroTest.java

```diff
@@ -536,4 +536,46 @@ private XDOM getXDOM(String content) throws Exception
 
         return blocks;
     }
+    
+    @Test
+    void executeIncludeMacroWhenExcludeFirstHeadingTrue() throws Exception 
```

Comment:
You're still missing at least 3-4 other tests... Try to think of all possible inputs and paths taken in the code you added.

Reply:
**Inputs:** True and false

**True:** covered
**False:** Yet to cover

**Paths:**
**1.** When heading is null(heading not found).
**2.** When heading is not null (heading found).
**3.** When heading is not the first heading.

So, there is a total of 5 test cases that need to be designed. I have covered one till now.
Please correct me if I am wrong.

Reply:
2 doesn't mean anything, it's not precise. Maybe you meant when the heading the first block in the include?

You're still missing the most important use cases that would show that your code is buggy... I dropped several hints already (more than hint, I already mentioned your code is not correct above)...

Reply:
3 is also not precise enough.

Reply:
Just write them and you'll see.

Reply:
You're on the right way